components/raftstore/src/store/snap/io.rs

```diff
@@ -202,22 +248,23 @@ where
             }
         }
 
-        while entry_len > remained_quota {
-            // It's possible to acquire more than necessary, but let it be.
-            io_limiter.blocking_consume(IO_LIMITER_CHUNK_SIZE);
-            remained_quota += IO_LIMITER_CHUNK_SIZE;
-        }
-        remained_quota -= entry_len;
-
         stats.key_count += 1;
         stats.total_size += entry_len;
+
+        if stats.total_size >= next_io_check_size {
+            handle_read_io_usage(&mut prev_io_bytes, &mut remained_quota);
+            next_io_check_size = stats.total_size + SCAN_BYTES_PER_IO_LIMIT_CHECK;
+        }
+
         if let Err(e) = sst_writer.borrow_mut().put(key, value) {
             let io_error = io::Error::new(io::ErrorKind::Other, e);
             return Err(io_error.into());
         }
         file_length += entry_len;
         Ok(true)
     }));
+    handle_read_io_usage(&mut prev_io_bytes, &mut remained_quota);
```

Comment:
nit: recommend to add some annotations or statements here, to specify that the last call of `handle_read_io_usage` is used to calibrate the final part of Read I/O cost.

Reply:
done